### PR TITLE
[SGTM fix dynamodb lock] refactor lock behaviour and tests and test utils

### PR DESCRIPTION
**Pull request by suzyng83209**
### Summary

I've noticed increasing lambda duration times for sgtm because of the long retry_timeout on our lock acquitision so
- instead of throwing on lock timeouts and retrying, we just accept that these are no-ops and add it back to the queue
- also shorten the timeout to the default (heartbeat_period + lease_duration = ~25s)
- also adding a `TEST` environment so running tests is easier and less noisy

Asana tasks:
https://app.asana.com/0/1207092173713920/1207404207836406/f

Relevant deployment: 

CC: @vn6 @harshita-gupta @prebeta @suzyng83209 @michael-huang87

### Test Plan
- [x] tests pass on this PR

### Risks

Pull Request: https://github.com/Asana/SGTM/pull/186

Pull Request synchronized with [Asana task](https://app.asana.com/0/0/1207458121014254)

### CODE REVIEW

File: src/github/webhook.py [hunk 1]
```diff
@@ -192,4 +193,8 @@ def handle_github_webhook(event_type, payload) -> HttpResponse:
     # if Github's data consistency needs a bit of time (does not have
     # read-after-write consistency)
     time.sleep(2)
-    return _events_map[event_type](payload)
+    try:
+        return _events_map[event_type](payload)
+    except DynamoDBLockError as e:
```
Comment (suzyng83209) [inline src/github/webhook.py:198]: alternatively, we could throw these errors and handle them as part of the queue retry logic. But I believe that's currently turned off?

**Comment by suzyng83209** [comment]
* **#186** <a href="https://app.graphite.dev/github/pr/Asana/SGTM/186?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @suzyng83209 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/config.py [hunk 2]
```diff
@@ -82,3 +76,15 @@ def is_feature_flag_enabled(flag_name: str) -> bool:
     ASANA_API_KEY = keys.get("ASANA_API_KEY", "")
     GITHUB_API_KEY = keys.get("GITHUB_API_KEY", "")
     GITHUB_HMAC_SECRET = keys.get("GITHUB_HMAC_SECRET", "")
+elif os.getenv("TEST"):
```
Comment (harshita-gupta) [inline src/config.py:79]: nice

File: src/dynamodb/lock.py [hunk 3]
```diff
@@ -11,22 +9,10 @@
 lock_client = DynamoDBLockClient(
     dynamodb_resource,
     table_name=LOCK_TABLE,
+    lease_duration=timedelta(seconds=20),
```
Comment (harshita-gupta) [inline src/dynamodb/lock.py:12]: oooh spicy

File: src/github/webhook.py (hunk 1 above)
Comment (vn6) [reply src/github/webhook.py:198]: queue logic is actually back on! so we could throw these

File: src/asana/helpers.py [hunk 4]
```diff
@@ -128,10 +128,9 @@ def _custom_fields_from_pull_request(pull_request: PullRequest) -> Dict:
     project_id = dynamodb_client.get_asana_id_from_github_node_id(repository_id)
 
     if project_id is None:
-        logger.info(
+        logger.error(
             f"Task not found for pull request {pull_request.id()}. Running a full sync!"
```
Comment (vn6) [inline src/asana/helpers.py:132]: nit: 
```suggestion
            f"Task not found for pull request {pull_request.id()}."
```

File: src/dynamodb/client.py [hunk 5]
```diff
@@ -121,17 +119,7 @@ def bulk_insert_github_node_to_asana_id_mapping(
...
-            return boto3.client("dynamodb")
-        except NoRegionError:
-            pass
-        # by raising the new error outside of the except clause, the ConfigurationError does not automatically contain
-        # the stack trace of the NoRegionError, which provides no extra value and clutters the console.
-        raise ConfigurationError(
-            "Configuration error: Please select a region, e.g. via"
-            " `AWS_DEFAULT_REGION=us-east-1`"
-        )
+        return boto3.client("dynamodb", region_name=AWS_REGION)
```
Comment (vn6) [inline src/dynamodb/client.py:122]: nice :)

File: src/dynamodb/lock.py (hunk 3 above)
Comment (vn6) [reply src/dynamodb/lock.py:12]: hmm do we have any way to log if the lease expires but the lock hasn't been released yet, might be helpful for debugging if this is too short

**Review by vn6 — APPROVED** [review]
lgtm! some small comments but thanks for cleaning things up!!

**Comment by suzyng83209** [comment]
@vn6 Assigning back for another passover. I ended up re-implementing the retry for DynamodbLockErrors and had to move the test into a new file `test_handler.py` which spawned a new mock test base to mock sqs operations.

File: test/test_handler.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            mock.side_effect = Exception("error")
+            response = handle_github_webhook(
+                event_type="pull_request",
+                delivery_id="123",
+                webhook_body=json.dumps(WEBHOOK_BODY_TEMPLATE),
+                should_retry=False,
+            )
+
+        self.assertEqual(response["statusCode"], "500")
+        self.assertEqual(response["body"], "error")
```
Comment (vn6) [inline test/test_handler.py:56]: should we also add a test case for no errors?

**Review by vn6 — APPROVED** [review]
lgtm!

**Pull request merged**